clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -207,19 +214,23 @@ private StorageOperationType getStorageOperationType(int partition, ByteBuffer v
       logStorageOperationWhileUnsubscribed(partition);
       return StorageOperationType.NONE;
     }
-    if (valuePayload != null && valuePayload.remaining() == 0 && rmdPayload.remaining() != 0) {
-      return StorageOperationType.RMD;
+    if (isDaVinciClient) {
+      return StorageOperationType.VALUE;
     }
-    if ((pcs.isEndOfPushReceived() || rmdPayload.remaining() != 0) && !isDaVinciClient) {
-      return StorageOperationType.VALUE_AND_RMD;
+    if (pcs.isEndOfPushReceived() || rmdPayload.remaining() > 0) {
+      if (valuePayload == null || valuePayload.remaining() > 0) {
```

Comment:
Yeah it should be always there as if you see VW, it always put default for RMD and RMD protocol ID. I can add the safety check just in case for regression.